rudder-server-relay/SOURCES/relay-api/relay_api/views.py

```diff
@@ -45,6 +45,22 @@ def put_file(target_uuid, source_uuid, file_id):
     return format_error(e, API_DEBUGINFO)
 
 
+@app.route("/shared-folder/<path:file_name>", methods=["HEAD"])
+def head_shared_folder(file_name):
+  try:
+    hash_type = request.args["hash_type"]
+    if hash_type is None:
+      hash_type = "sha256"
+    file_hash = request.args["hash"]
+    if file_hash is None:
+      file_hash = ""
```

Comment:
It will fail, but i choose to delegate all error checking in the method from shared-files and stay as simple as possible.

We can fail directly if you prefer